src/ray/core_worker/core_worker.cc

```diff
@@ -2010,18 +2010,16 @@ Status CoreWorker::Contains(const ObjectID &object_id,
 
 // For any objects that are ErrorType::OBJECT_IN_PLASMA, we need to move them from
 // the ready set into the plasma_object_ids set to wait on them there.
-void RetryObjectInPlasmaErrors(std::shared_ptr<CoreWorkerMemoryStore> &memory_store,
-                               WorkerContext &worker_context,
+void MoveReadyPlasmaObjectsToPlasmaSet(std::shared_ptr<CoreWorkerMemoryStore> &memory_store,
                                absl::flat_hash_set<ObjectID> &memory_object_ids,
                                absl::flat_hash_set<ObjectID> &plasma_object_ids,
                                absl::flat_hash_set<ObjectID> &ready) {
-  for (auto iter = memory_object_ids.begin(); iter != memory_object_ids.end();) {
-    auto current = iter++;
-    const auto &mem_id = *current;
-    auto found = memory_store->GetIfExists(mem_id);
+  for (auto iter = ready.begin(); iter != ready.end(); iter++) {
+    const auto &obj_id = *iter;
+    auto found = memory_store->GetIfExists(obj_id);
     if (found != nullptr && found->IsInPlasmaError()) {
-      plasma_object_ids.insert(mem_id);
-      ready.erase(mem_id);
+      plasma_object_ids.insert(obj_id);
+      ready.erase(obj_id);
       memory_object_ids.erase(current);
```

Comment:
This needs to be updated to `obj_id`, right?

Reply:
```suggestion
      memory_object_ids.erase(iter);
```
actually my bad from the suggestion i applied